ntbs-service/Services/AzureAdDirectoryService.cs

```diff
@@ -141,45 +141,51 @@ private async Task<IEnumerable<Microsoft.Graph.User>> GetAllDirectoryEntries()
             var baseGroups = await _graphServiceClient
             .Groups
             .Request()
-            .Filter($"startswith(displayName, '{_adOptions.BaseUserGroup}')")
+            .Filter($"displayName eq '{_adOptions.BaseUserGroup}'")
             .Select("id, displayName, description")
-            .Top(999)
+            .Top(1)
             .GetAsync();
 
             var baseGroup = baseGroups.FirstOrDefault(g => g.DisplayName == _adOptions.BaseUserGroup);
 
             if (baseGroup != null)
             {
+                var usersInRootGroup = await GetUsersInGroup(baseGroup.Id);
+                listOfDirectoryEntries.AddRange(usersInRootGroup);
 
-                var groupMembers = await _graphServiceClient.Groups[baseGroup.Id]
-                    .Members
-                    .Request()
-                    .GetAsync();
-
-                if (groupMembers != null)
+                var groupMembersRequest = _graphServiceClient.Groups[baseGroup.Id].Members.Request();
+                while (groupMembersRequest != null)
```

Comment:
There's a lot of nesting going on here - it could definitely be tidied up by breaking it up into a few different methods.

Up to you whether you want to do that now, or just leave it as future tech debt.

Reply:
I've had a go at refactoring this. I haven't actually added many more methods, but have reduced the nesting a fair amount